### PR TITLE
Simplify the krshaped interface, just use k8s interfaces for TypeMeta and ObjectMeta

### PR DESCRIPTION
**Pull request by whaught**
I had been using the GetObjectMeta accessor, but since these are inlined we can just use the metav1.Object interface directly.

Likewise we'll use the schema.ObjectKind interface to access TypeMeta information

### CODE REVIEW

File: apis/duck/v1/kresource_type.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ import (
 
 // KRShaped is an interface for retrieving the duck elements of an arbitrary resource.
 type KRShaped interface {
-	metav1.ObjectMetaAccessor
+	metav1.Object
 
 	GetTypeMeta() *metav1.TypeMeta
```
Comment (dprotaso) [inline apis/duck/v1/kresource_type.go:33]: I'd inline the metav1.TypeMeta methods we care about which should let you drop having downstream resources from implementing `GetTypeMeta()`

```suggestion
	SetGroupVersionKind(gvk schema.GroupVersionKind)
	GroupVersionKind() schema.GroupVersionKind
```

alternatively these satisfy the following interface

```suggestion
	schema.ObjectKind
```

File: apis/duck/v1/kresource_type.go [hunk 2]
```diff
@@ -22,15 +22,15 @@ import (
...
 	"knative.dev/pkg/apis"
 )
 
 // KRShaped is an interface for retrieving the duck elements of an arbitrary resource.
 type KRShaped interface {
-	metav1.ObjectMetaAccessor
-
-	GetTypeMeta() *metav1.TypeMeta
+	metav1.Object
+	schema.ObjectKind
```
Comment (mattmoor) [inline apis/duck/v1/kresource_type.go:33]: Remove trailing whitespace:
```suggestion
	schema.ObjectKind
```

Comment (mattmoor) [inline apis/duck/v1/kresource_type.go:33]: Format Go code:
```suggestion
	schema.ObjectKind
```

**Comment by mattmoor** [comment]
/lgtm
/approve

nice cleanup!

**Pull request merged**